TodoListServiceMT/Controllers/TodoListController.cs

```diff
@@ -36,18 +36,40 @@ public class TodoListController : ApiController
         // Returns a list of all the saved Todo items associated to the current caller
         public IEnumerable<Todo> Get()
         {
+            VerifyScope();
+
             string owner = ClaimsPrincipal.Current.FindFirst(ClaimTypes.NameIdentifier).Value;
             var currentUserToDos = db.Todoes.Where(a => a.Owner == owner);
             return (currentUserToDos.ToList());
         }
 
+
         // POST api/todoapi
         // Saves in the db a new Todo item associated to the current caller
         public void Post(Todo todo)
         {
+            VerifyScope();
+
             todo.Owner = ClaimsPrincipal.Current.FindFirst(ClaimTypes.NameIdentifier).Value;
             db.Todoes.Add(todo);
             db.SaveChanges();
         }
+
+        /// <summary>
+        /// Verifies that the JWT used to call the API:
+        /// 1) has user claims
+        /// 2) has a scope claim of value "user_impersonation".
+        /// If that's not the case an HTTP 401 is returned.
+        /// </summary>
+        private static void VerifyScope()
```

Comment:
Having a call to this method looks like an extra hassle. Could you work with the middleware team to explore the following two options
1. Carry this check right after token validation in startup.cs 
2. Extend the AuthorizeAttribute and move this check there.

Reply:
@kalyankrishna1 : in general, (1.) would not be very good as there could be different scopes to verify depending on the API.
(2) sounds promissing.

Reply:
For 1., IMHO we just need to ensure that we teach that the scope claim is present, because that proves consent took place. How, where and for what purpose the a scope is used is a developer's concern.